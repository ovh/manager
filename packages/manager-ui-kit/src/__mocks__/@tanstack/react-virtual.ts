import { vi } from 'vitest';

/**
 * Mock implementation of @tanstack/react-virtual `useVirtualizer`.
 * Simulates virtualization logic for predictable test behavior.
 */
export const useVirtualizer = vi.fn((options?: { count?: number; estimateSize?: () => number }) => {
  const count = options?.count ?? 0;
  const estimateSize = options?.estimateSize ?? (() => 50);

  // Build a fake list of virtual items
  const virtualItems = Array.from({ length: count }, (_, index) => {
    const size = estimateSize();
    const start = index * size;
    const end = start + size;
    return { index, start, size, end, key: index };
  });

  /** Mock API methods â€” keep them individually callable for assertions */
  const scrollToIndex = vi.fn();
  const measure = vi.fn();
  const getScrollElement = vi.fn(() => ({ scrollTop: 0 }));
  const getVirtualItems = vi.fn(() => virtualItems);
  const getTotalSize = vi.fn(() => virtualItems.reduce((acc, v) => acc + v.size, 0));

  return {
    getVirtualItems,
    getTotalSize,
    scrollToIndex,
    measure,
    getScrollElement,
  };
});
