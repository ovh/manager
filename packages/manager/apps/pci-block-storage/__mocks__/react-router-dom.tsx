import { vi } from 'vitest';

export * from 'react-router-dom';

export const useMatches = vi.fn(() => [{}]);

export const useParams = vi.fn(() => ({
  projectId: 'my-project-id',
}));

export const useHref = vi.fn((path) => path);

export const useHistory = vi.fn((path) => path);

export const Navigate = vi.fn(() => <></>);
