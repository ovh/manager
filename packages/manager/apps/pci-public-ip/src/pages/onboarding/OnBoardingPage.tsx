import {
  useEnvironment,
  useNavigation,
} from '@ovh-ux/manager-react-shell-client';
import {
  Card,
  OnboardingLayout,
  PciDiscoveryBanner,
  isDiscoveryProject,
} from '@ovhcloud/manager-components';
import { OsdsBreadcrumb, OsdsText } from '@ovhcloud/ods-components/react';
import { useEffect, useState } from 'react';
import { useTranslation } from 'react-i18next';
import {
  Outlet,
  useNavigate,
  useParams,
  useRouteLoaderData,
} from 'react-router-dom';
import {
  ODS_THEME_COLOR_INTENT,
  ODS_THEME_TYPOGRAPHY_SIZE,
} from '@ovhcloud/ods-common-theming';
import {
  ODS_TEXT_LEVEL,
  OdsBreadcrumbAttributeItem,
} from '@ovhcloud/ods-components';
import { Project } from '@/api/data/project';
import { useAllFloatingIP } from '@/api/hooks/useFloatingIP';
import { GUIDES } from './onboarding.constants';

export default function OnBoardingPage() {
  const { t } = useTranslation();
  const { t: tOnBoarding } = useTranslation('onboarding');
  const { projectId } = useParams();
  const navigation = useNavigation();
  const { ovhSubsidiary } = useEnvironment().getUser();
  const project = useRouteLoaderData('public-ips') as Project;
  const [urlProject, setUrlProject] = useState('');
  const navigate = useNavigate();
  const { data: floatingIPs, isLoading } = useAllFloatingIP(projectId);

  useEffect(() => {
    navigation
      .getURL('public-cloud', `#/pci/projects/${projectId}`, {})
      .then((data) => {
        setUrlProject(data as string);
      });
  }, [projectId, navigation]);

  useEffect(() => {
    // if there are users redirect to user listing page
    if (!isLoading && floatingIPs.length > 0) {
      navigate(`/pci/projects/${projectId}/public-ips`);
    }
  }, [isLoading, floatingIPs, navigate]);

  const breadcrumbItems: OdsBreadcrumbAttributeItem[] = [
    {
      href: urlProject,
      label: project?.description,
    },
    {
      label: t('pci_additional_ips_title'),
    },
  ];

  const tileItems = [
    {
      id: 'create-delete-openstack-user',
      texts: {
        title: tOnBoarding(
          'pci_additional_ips_onboarding_guides_additional_public_ips_title',
        ),
        description: tOnBoarding(
          'pci_additional_ips_onboarding_guides_additional_public_ips_description',
        ),
        category: tOnBoarding('onboarding_guide_title'),
      },
      href: GUIDES.additional_public_ips[ovhSubsidiary],
    },
    {
      id: 'configure-horizon-user-access',
      texts: {
        title: tOnBoarding(
          'pci_additional_ips_onboarding_guides_configure_additional_ip_title',
        ),
        description: tOnBoarding(
          'pci_additional_ips_onboarding_guides_configure_additional_ip_description',
        ),
        category: tOnBoarding('onboarding_guide_title'),
      },
      href: GUIDES.configure_additional_ip[ovhSubsidiary],
    },
    {
      id: 'prepare-openstack-api-env',
      texts: {
        title: tOnBoarding(
          'pci_additional_ips_onboarding_guides_configure_floating_ip_title',
        ),
        description: tOnBoarding(
          'pci_additional_ips_onboarding_guides_configure_floating_ip_description',
        ),
        category: tOnBoarding('onboarding_guide_title'),
      },
      href: GUIDES.configure_floating_ip[ovhSubsidiary],
    },
  ];

  return (
    <>
      {project && <OsdsBreadcrumb items={breadcrumbItems} />}
      {isDiscoveryProject(project) && (
        <div className="mb-8">
          <PciDiscoveryBanner projectId={projectId} />
        </div>
      )}
      <OnboardingLayout
        title={t('pci_additional_ips_title')}
        description={
          <>
            <OsdsText
              color={ODS_THEME_COLOR_INTENT.text}
              level={ODS_TEXT_LEVEL.body}
              size={ODS_THEME_TYPOGRAPHY_SIZE._500}
            >
              {tOnBoarding('pci_additional_ips_onboarding_content1')}
            </OsdsText>
            <OsdsText
              color={ODS_THEME_COLOR_INTENT.text}
              level={ODS_TEXT_LEVEL.body}
              size={ODS_THEME_TYPOGRAPHY_SIZE._400}
              className="mt-4 block"
            >
              {tOnBoarding('pci_additional_ips_onboarding_content2')}
            </OsdsText>
            <OsdsText
              color={ODS_THEME_COLOR_INTENT.text}
              level={ODS_TEXT_LEVEL.body}
              size={ODS_THEME_TYPOGRAPHY_SIZE._400}
              className="mt-6 block"
            >
              {tOnBoarding('pci_additional_ips_onboarding_content3')}
            </OsdsText>
            <OsdsText
              color={ODS_THEME_COLOR_INTENT.text}
              level={ODS_TEXT_LEVEL.body}
              size={ODS_THEME_TYPOGRAPHY_SIZE._500}
              className="block mt-6"
            >
              {tOnBoarding('pci_additional_ips_onboarding_content4_heading')}
            </OsdsText>
            <OsdsText
              color={ODS_THEME_COLOR_INTENT.text}
              level={ODS_TEXT_LEVEL.body}
              size={ODS_THEME_TYPOGRAPHY_SIZE._400}
              className="block"
            >
              {tOnBoarding(
                'pci_additional_ips_onboarding_content4_description',
              )}
            </OsdsText>
            <OsdsText
              color={ODS_THEME_COLOR_INTENT.text}
              level={ODS_TEXT_LEVEL.body}
              size={ODS_THEME_TYPOGRAPHY_SIZE._500}
              className="block mt-6"
            >
              {tOnBoarding('pci_additional_ips_onboarding_content5_heading')}
            </OsdsText>
            <OsdsText
              color={ODS_THEME_COLOR_INTENT.text}
              level={ODS_TEXT_LEVEL.body}
              size={ODS_THEME_TYPOGRAPHY_SIZE._400}
              className="block"
            >
              {tOnBoarding(
                'pci_additional_ips_onboarding_content5_description',
              )}
            </OsdsText>
          </>
        }
        orderButtonLabel={tOnBoarding(
          'pci_additional_ips_onboarding_action_buy',
        )}
        onOrderButtonClick={() => {
          if (!isDiscoveryProject(project)) navigate('./new');
        }}
        isActionDisabled={isDiscoveryProject(project)}
      >
        <aside className="grid grid-cols-1 sm:grid-cols-2 md:grid-cols-3 gap-8 pt-12">
          {tileItems.map((tile) => (
            <Card key={tile.id} href={tile.href} texts={tile.texts} />
          ))}
        </aside>
      </OnboardingLayout>
      <Outlet />
    </>
  );
}
