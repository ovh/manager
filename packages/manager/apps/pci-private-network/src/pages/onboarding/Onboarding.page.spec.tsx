import 'element-internals-polyfill';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import { describe, expect, vi } from 'vitest';
import { render, waitFor } from '@testing-library/react';
import '@testing-library/jest-dom';
import {
  ShellContext,
  ShellContextType,
} from '@ovh-ux/manager-react-shell-client';
import { useParams } from 'react-router-dom';
import OnBoardingPage from './Onboarding.page';

vi.mock('react-router-dom');

vi.mock('./OnBoardingGuard', () => ({
  default: ({ children }) => <>{children}</>,
}));

const shellContext = {
  environment: {
    getUser: vi.fn(),
  },
  shell: {
    navigation: {
      getURL: vi.fn(),
    },
  },
};

const queryClient = new QueryClient();
const wrapper = ({ children }) => (
  <QueryClientProvider client={queryClient}>
    <ShellContext.Provider
      value={(shellContext as unknown) as ShellContextType}
    >
      {children}
    </ShellContext.Provider>
  </QueryClientProvider>
);

describe('OnBoardingPage', () => {
  it('should render onboarding page', async () => {
    const { environment, shell } = shellContext;
    environment.getUser.mockResolvedValue({
      ovhSubsidiary: 'foo',
    });
    shell.navigation.getURL.mockResolvedValue('https://www.ovh.com');
    vi.mocked(useParams).mockReturnValue({ projectId: '123' });
    const { container } = render(<OnBoardingPage />, { wrapper });
    await waitFor(() => {
      expect(container).toBeDefined();
    });
  });
});
