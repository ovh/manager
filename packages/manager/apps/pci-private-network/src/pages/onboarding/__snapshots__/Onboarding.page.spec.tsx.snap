// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`OnBoardingPage > displays private network creation button if vrack exists on project 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex flex-col mx-auto sm:px-10"
      >
        <section
          class="flex flex-col items-center mt-8"
        >
          <img
            class="max-h-28 "
            src="../../../manager-react-components/src/components/templates/onboarding/assets/placeholder.png"
          />
          <osds-text
            class="block text-center sm:pt-8 xs:pt-2.5"
            color="primary"
            hue="800"
            level="heading"
            size="700"
          >
            pci_projects_project_network_private
          </osds-text>
          <osds-text
            class="block text-center xs:pt-2.5 sm:pt-8 max-w-4xl"
            color="text"
            level="body"
            size="400"
          >
            <osds-text
              color="text"
              level="body"
              size="400"
            >
              pci_projects_project_network_private_vrack_empty
            </osds-text>
            <osds-text
              class="mt-6 block"
              color="text"
              level="body"
              size="500"
            >
              pci_projects_project_network_private_vrack_deploy
            </osds-text>
            <osds-text
              class="mt-4 block"
              color="text"
              level="body"
              size="400"
            >
              pci_projects_project_network_private_vrack_explanation_1
            </osds-text>
            <osds-text
              class="mt-6 block"
              color="text"
              level="body"
              size="400"
            >
              pci_projects_project_network_private_vrack_explanation_2
            </osds-text>
          </osds-text>
          <div
            class="flex sm:pt-8 xs:pt-2.5 flex-row items-center space-x-4 justify-center"
          >
            <osds-button
              color="primary"
              inline="true"
              size="md"
            >
              pci_projects_project_network_private_create
            </osds-button>
          </div>
        </section>
        <aside
          class="grid grid-cols-1 sm:grid-cols-2 md:grid-cols-3 gap-6 xs:pt-10 sm:pt-20"
        >
          <a
            class="no-underline"
            href="https://docs.ovh.com/gb/en/publiccloud/network-services/creating-private-network-with-gateway/"
            target="_blank"
          >
            <osds-tile
              class="w-full h-full"
              color="primary"
              inline="true"
              rounded="true"
              variant="stroked"
            >
              <div
                class="flex flex-col "
              >
                <osds-text
                  class="block"
                  color="primary"
                  hue="500"
                  level="heading"
                  size="400"
                >
                  onboarding_guide_title
                  <span />
                </osds-text>
                <osds-text
                  class="block mb-4"
                  color="primary"
                  hue="800"
                  level="heading"
                  size="500"
                >
                  pci_projects_project_network_private_vrack_guides_create_private_network_with_gateway_title
                </osds-text>
                <osds-link
                  color="primary"
                  tab-index="-1"
                >
                  <span
                    slot="start"
                  />
                  see_more_label
                  <span
                    slot="end"
                  >
                    <osds-icon
                      aria-hidden="true"
                      class="ml-4"
                      color="primary"
                      hoverable="true"
                      name="arrow-right"
                      size="xxs"
                    />
                  </span>
                </osds-link>
              </div>
            </osds-tile>
          </a>
          <a
            class="no-underline"
            href="https://docs.ovh.com/gb/en/public-cloud/attaching-pci-floating-ip-to-instance/"
            target="_blank"
          >
            <osds-tile
              class="w-full h-full"
              color="primary"
              inline="true"
              rounded="true"
              variant="stroked"
            >
              <div
                class="flex flex-col "
              >
                <osds-text
                  class="block"
                  color="primary"
                  hue="500"
                  level="heading"
                  size="400"
                >
                  onboarding_guide_title
                  <span />
                </osds-text>
                <osds-text
                  class="block mb-4"
                  color="primary"
                  hue="800"
                  level="heading"
                  size="500"
                >
                  pci_projects_project_network_private_vrack_guides_attach_pci_floating_ip_to_instance_title
                </osds-text>
                <osds-link
                  color="primary"
                  tab-index="-1"
                >
                  <span
                    slot="start"
                  />
                  see_more_label
                  <span
                    slot="end"
                  >
                    <osds-icon
                      aria-hidden="true"
                      class="ml-4"
                      color="primary"
                      hoverable="true"
                      name="arrow-right"
                      size="xxs"
                    />
                  </span>
                </osds-link>
              </div>
            </osds-tile>
          </a>
          <a
            class="no-underline"
            href="https://www.ovhcloud.com/en-gb/public-cloud/private-network"
            target="_blank"
          >
            <osds-tile
              class="w-full h-full"
              color="primary"
              inline="true"
              rounded="true"
              variant="stroked"
            >
              <div
                class="flex flex-col "
              >
                <osds-text
                  class="block"
                  color="primary"
                  hue="500"
                  level="heading"
                  size="400"
                >
                  onboarding_guide_title
                  <span />
                </osds-text>
                <osds-text
                  class="block mb-4"
                  color="primary"
                  hue="800"
                  level="heading"
                  size="500"
                >
                  pci_projects_project_network_private_vrack_guides_private_networks_title
                </osds-text>
                <osds-link
                  color="primary"
                  tab-index="-1"
                >
                  <span
                    slot="start"
                  />
                  see_more_label
                  <span
                    slot="end"
                  >
                    <osds-icon
                      aria-hidden="true"
                      class="ml-4"
                      color="primary"
                      hoverable="true"
                      name="arrow-right"
                      size="xxs"
                    />
                  </span>
                </osds-link>
              </div>
            </osds-tile>
          </a>
        </aside>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex flex-col mx-auto sm:px-10"
    >
      <section
        class="flex flex-col items-center mt-8"
      >
        <img
          class="max-h-28 "
          src="../../../manager-react-components/src/components/templates/onboarding/assets/placeholder.png"
        />
        <osds-text
          class="block text-center sm:pt-8 xs:pt-2.5"
          color="primary"
          hue="800"
          level="heading"
          size="700"
        >
          pci_projects_project_network_private
        </osds-text>
        <osds-text
          class="block text-center xs:pt-2.5 sm:pt-8 max-w-4xl"
          color="text"
          level="body"
          size="400"
        >
          <osds-text
            color="text"
            level="body"
            size="400"
          >
            pci_projects_project_network_private_vrack_empty
          </osds-text>
          <osds-text
            class="mt-6 block"
            color="text"
            level="body"
            size="500"
          >
            pci_projects_project_network_private_vrack_deploy
          </osds-text>
          <osds-text
            class="mt-4 block"
            color="text"
            level="body"
            size="400"
          >
            pci_projects_project_network_private_vrack_explanation_1
          </osds-text>
          <osds-text
            class="mt-6 block"
            color="text"
            level="body"
            size="400"
          >
            pci_projects_project_network_private_vrack_explanation_2
          </osds-text>
        </osds-text>
        <div
          class="flex sm:pt-8 xs:pt-2.5 flex-row items-center space-x-4 justify-center"
        >
          <osds-button
            color="primary"
            inline="true"
            size="md"
          >
            pci_projects_project_network_private_create
          </osds-button>
        </div>
      </section>
      <aside
        class="grid grid-cols-1 sm:grid-cols-2 md:grid-cols-3 gap-6 xs:pt-10 sm:pt-20"
      >
        <a
          class="no-underline"
          href="https://docs.ovh.com/gb/en/publiccloud/network-services/creating-private-network-with-gateway/"
          target="_blank"
        >
          <osds-tile
            class="w-full h-full"
            color="primary"
            inline="true"
            rounded="true"
            variant="stroked"
          >
            <div
              class="flex flex-col "
            >
              <osds-text
                class="block"
                color="primary"
                hue="500"
                level="heading"
                size="400"
              >
                onboarding_guide_title
                <span />
              </osds-text>
              <osds-text
                class="block mb-4"
                color="primary"
                hue="800"
                level="heading"
                size="500"
              >
                pci_projects_project_network_private_vrack_guides_create_private_network_with_gateway_title
              </osds-text>
              <osds-link
                color="primary"
                tab-index="-1"
              >
                <span
                  slot="start"
                />
                see_more_label
                <span
                  slot="end"
                >
                  <osds-icon
                    aria-hidden="true"
                    class="ml-4"
                    color="primary"
                    hoverable="true"
                    name="arrow-right"
                    size="xxs"
                  />
                </span>
              </osds-link>
            </div>
          </osds-tile>
        </a>
        <a
          class="no-underline"
          href="https://docs.ovh.com/gb/en/public-cloud/attaching-pci-floating-ip-to-instance/"
          target="_blank"
        >
          <osds-tile
            class="w-full h-full"
            color="primary"
            inline="true"
            rounded="true"
            variant="stroked"
          >
            <div
              class="flex flex-col "
            >
              <osds-text
                class="block"
                color="primary"
                hue="500"
                level="heading"
                size="400"
              >
                onboarding_guide_title
                <span />
              </osds-text>
              <osds-text
                class="block mb-4"
                color="primary"
                hue="800"
                level="heading"
                size="500"
              >
                pci_projects_project_network_private_vrack_guides_attach_pci_floating_ip_to_instance_title
              </osds-text>
              <osds-link
                color="primary"
                tab-index="-1"
              >
                <span
                  slot="start"
                />
                see_more_label
                <span
                  slot="end"
                >
                  <osds-icon
                    aria-hidden="true"
                    class="ml-4"
                    color="primary"
                    hoverable="true"
                    name="arrow-right"
                    size="xxs"
                  />
                </span>
              </osds-link>
            </div>
          </osds-tile>
        </a>
        <a
          class="no-underline"
          href="https://www.ovhcloud.com/en-gb/public-cloud/private-network"
          target="_blank"
        >
          <osds-tile
            class="w-full h-full"
            color="primary"
            inline="true"
            rounded="true"
            variant="stroked"
          >
            <div
              class="flex flex-col "
            >
              <osds-text
                class="block"
                color="primary"
                hue="500"
                level="heading"
                size="400"
              >
                onboarding_guide_title
                <span />
              </osds-text>
              <osds-text
                class="block mb-4"
                color="primary"
                hue="800"
                level="heading"
                size="500"
              >
                pci_projects_project_network_private_vrack_guides_private_networks_title
              </osds-text>
              <osds-link
                color="primary"
                tab-index="-1"
              >
                <span
                  slot="start"
                />
                see_more_label
                <span
                  slot="end"
                >
                  <osds-icon
                    aria-hidden="true"
                    class="ml-4"
                    color="primary"
                    hoverable="true"
                    name="arrow-right"
                    size="xxs"
                  />
                </span>
              </osds-link>
            </div>
          </osds-tile>
        </a>
      </aside>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`OnBoardingPage > displays private vrack creation button if missing vrack on project 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex flex-col mx-auto sm:px-10"
      >
        <section
          class="flex flex-col items-center mt-8"
        >
          <img
            class="max-h-28 "
            src="../../../manager-react-components/src/components/templates/onboarding/assets/placeholder.png"
          />
          <osds-text
            class="block text-center sm:pt-8 xs:pt-2.5"
            color="primary"
            hue="800"
            level="heading"
            size="700"
          >
            pci_projects_project_network_private
          </osds-text>
          <osds-text
            class="block text-center xs:pt-2.5 sm:pt-8 max-w-4xl"
            color="text"
            level="body"
            size="400"
          >
            <osds-text
              color="text"
              level="body"
              size="400"
            >
              pci_projects_project_network_private_vrack_empty
            </osds-text>
            <osds-text
              class="mt-6 block"
              color="text"
              level="body"
              size="500"
            >
              pci_projects_project_network_private_vrack_deploy
            </osds-text>
            <osds-text
              class="mt-4 block"
              color="text"
              level="body"
              size="400"
            >
              pci_projects_project_network_private_vrack_explanation_1
            </osds-text>
            <osds-text
              class="mt-6 block"
              color="text"
              level="body"
              size="400"
            >
              pci_projects_project_network_private_vrack_explanation_2
            </osds-text>
          </osds-text>
          <div
            class="flex sm:pt-8 xs:pt-2.5 flex-row items-center space-x-4 justify-center"
          >
            <osds-button
              color="primary"
              inline="true"
              size="md"
            >
              pci_projects_project_network_private_vrack_create_heading
            </osds-button>
          </div>
        </section>
        <aside
          class="grid grid-cols-1 sm:grid-cols-2 md:grid-cols-3 gap-6 xs:pt-10 sm:pt-20"
        >
          <a
            class="no-underline"
            href="https://docs.ovh.com/gb/en/publiccloud/network-services/creating-private-network-with-gateway/"
            target="_blank"
          >
            <osds-tile
              class="w-full h-full"
              color="primary"
              inline="true"
              rounded="true"
              variant="stroked"
            >
              <div
                class="flex flex-col "
              >
                <osds-text
                  class="block"
                  color="primary"
                  hue="500"
                  level="heading"
                  size="400"
                >
                  onboarding_guide_title
                  <span />
                </osds-text>
                <osds-text
                  class="block mb-4"
                  color="primary"
                  hue="800"
                  level="heading"
                  size="500"
                >
                  pci_projects_project_network_private_vrack_guides_create_private_network_with_gateway_title
                </osds-text>
                <osds-link
                  color="primary"
                  tab-index="-1"
                >
                  <span
                    slot="start"
                  />
                  see_more_label
                  <span
                    slot="end"
                  >
                    <osds-icon
                      aria-hidden="true"
                      class="ml-4"
                      color="primary"
                      hoverable="true"
                      name="arrow-right"
                      size="xxs"
                    />
                  </span>
                </osds-link>
              </div>
            </osds-tile>
          </a>
          <a
            class="no-underline"
            href="https://docs.ovh.com/gb/en/public-cloud/attaching-pci-floating-ip-to-instance/"
            target="_blank"
          >
            <osds-tile
              class="w-full h-full"
              color="primary"
              inline="true"
              rounded="true"
              variant="stroked"
            >
              <div
                class="flex flex-col "
              >
                <osds-text
                  class="block"
                  color="primary"
                  hue="500"
                  level="heading"
                  size="400"
                >
                  onboarding_guide_title
                  <span />
                </osds-text>
                <osds-text
                  class="block mb-4"
                  color="primary"
                  hue="800"
                  level="heading"
                  size="500"
                >
                  pci_projects_project_network_private_vrack_guides_attach_pci_floating_ip_to_instance_title
                </osds-text>
                <osds-link
                  color="primary"
                  tab-index="-1"
                >
                  <span
                    slot="start"
                  />
                  see_more_label
                  <span
                    slot="end"
                  >
                    <osds-icon
                      aria-hidden="true"
                      class="ml-4"
                      color="primary"
                      hoverable="true"
                      name="arrow-right"
                      size="xxs"
                    />
                  </span>
                </osds-link>
              </div>
            </osds-tile>
          </a>
          <a
            class="no-underline"
            href="https://www.ovhcloud.com/en-gb/public-cloud/private-network"
            target="_blank"
          >
            <osds-tile
              class="w-full h-full"
              color="primary"
              inline="true"
              rounded="true"
              variant="stroked"
            >
              <div
                class="flex flex-col "
              >
                <osds-text
                  class="block"
                  color="primary"
                  hue="500"
                  level="heading"
                  size="400"
                >
                  onboarding_guide_title
                  <span />
                </osds-text>
                <osds-text
                  class="block mb-4"
                  color="primary"
                  hue="800"
                  level="heading"
                  size="500"
                >
                  pci_projects_project_network_private_vrack_guides_private_networks_title
                </osds-text>
                <osds-link
                  color="primary"
                  tab-index="-1"
                >
                  <span
                    slot="start"
                  />
                  see_more_label
                  <span
                    slot="end"
                  >
                    <osds-icon
                      aria-hidden="true"
                      class="ml-4"
                      color="primary"
                      hoverable="true"
                      name="arrow-right"
                      size="xxs"
                    />
                  </span>
                </osds-link>
              </div>
            </osds-tile>
          </a>
        </aside>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex flex-col mx-auto sm:px-10"
    >
      <section
        class="flex flex-col items-center mt-8"
      >
        <img
          class="max-h-28 "
          src="../../../manager-react-components/src/components/templates/onboarding/assets/placeholder.png"
        />
        <osds-text
          class="block text-center sm:pt-8 xs:pt-2.5"
          color="primary"
          hue="800"
          level="heading"
          size="700"
        >
          pci_projects_project_network_private
        </osds-text>
        <osds-text
          class="block text-center xs:pt-2.5 sm:pt-8 max-w-4xl"
          color="text"
          level="body"
          size="400"
        >
          <osds-text
            color="text"
            level="body"
            size="400"
          >
            pci_projects_project_network_private_vrack_empty
          </osds-text>
          <osds-text
            class="mt-6 block"
            color="text"
            level="body"
            size="500"
          >
            pci_projects_project_network_private_vrack_deploy
          </osds-text>
          <osds-text
            class="mt-4 block"
            color="text"
            level="body"
            size="400"
          >
            pci_projects_project_network_private_vrack_explanation_1
          </osds-text>
          <osds-text
            class="mt-6 block"
            color="text"
            level="body"
            size="400"
          >
            pci_projects_project_network_private_vrack_explanation_2
          </osds-text>
        </osds-text>
        <div
          class="flex sm:pt-8 xs:pt-2.5 flex-row items-center space-x-4 justify-center"
        >
          <osds-button
            color="primary"
            inline="true"
            size="md"
          >
            pci_projects_project_network_private_vrack_create_heading
          </osds-button>
        </div>
      </section>
      <aside
        class="grid grid-cols-1 sm:grid-cols-2 md:grid-cols-3 gap-6 xs:pt-10 sm:pt-20"
      >
        <a
          class="no-underline"
          href="https://docs.ovh.com/gb/en/publiccloud/network-services/creating-private-network-with-gateway/"
          target="_blank"
        >
          <osds-tile
            class="w-full h-full"
            color="primary"
            inline="true"
            rounded="true"
            variant="stroked"
          >
            <div
              class="flex flex-col "
            >
              <osds-text
                class="block"
                color="primary"
                hue="500"
                level="heading"
                size="400"
              >
                onboarding_guide_title
                <span />
              </osds-text>
              <osds-text
                class="block mb-4"
                color="primary"
                hue="800"
                level="heading"
                size="500"
              >
                pci_projects_project_network_private_vrack_guides_create_private_network_with_gateway_title
              </osds-text>
              <osds-link
                color="primary"
                tab-index="-1"
              >
                <span
                  slot="start"
                />
                see_more_label
                <span
                  slot="end"
                >
                  <osds-icon
                    aria-hidden="true"
                    class="ml-4"
                    color="primary"
                    hoverable="true"
                    name="arrow-right"
                    size="xxs"
                  />
                </span>
              </osds-link>
            </div>
          </osds-tile>
        </a>
        <a
          class="no-underline"
          href="https://docs.ovh.com/gb/en/public-cloud/attaching-pci-floating-ip-to-instance/"
          target="_blank"
        >
          <osds-tile
            class="w-full h-full"
            color="primary"
            inline="true"
            rounded="true"
            variant="stroked"
          >
            <div
              class="flex flex-col "
            >
              <osds-text
                class="block"
                color="primary"
                hue="500"
                level="heading"
                size="400"
              >
                onboarding_guide_title
                <span />
              </osds-text>
              <osds-text
                class="block mb-4"
                color="primary"
                hue="800"
                level="heading"
                size="500"
              >
                pci_projects_project_network_private_vrack_guides_attach_pci_floating_ip_to_instance_title
              </osds-text>
              <osds-link
                color="primary"
                tab-index="-1"
              >
                <span
                  slot="start"
                />
                see_more_label
                <span
                  slot="end"
                >
                  <osds-icon
                    aria-hidden="true"
                    class="ml-4"
                    color="primary"
                    hoverable="true"
                    name="arrow-right"
                    size="xxs"
                  />
                </span>
              </osds-link>
            </div>
          </osds-tile>
        </a>
        <a
          class="no-underline"
          href="https://www.ovhcloud.com/en-gb/public-cloud/private-network"
          target="_blank"
        >
          <osds-tile
            class="w-full h-full"
            color="primary"
            inline="true"
            rounded="true"
            variant="stroked"
          >
            <div
              class="flex flex-col "
            >
              <osds-text
                class="block"
                color="primary"
                hue="500"
                level="heading"
                size="400"
              >
                onboarding_guide_title
                <span />
              </osds-text>
              <osds-text
                class="block mb-4"
                color="primary"
                hue="800"
                level="heading"
                size="500"
              >
                pci_projects_project_network_private_vrack_guides_private_networks_title
              </osds-text>
              <osds-link
                color="primary"
                tab-index="-1"
              >
                <span
                  slot="start"
                />
                see_more_label
                <span
                  slot="end"
                >
                  <osds-icon
                    aria-hidden="true"
                    class="ml-4"
                    color="primary"
                    hoverable="true"
                    name="arrow-right"
                    size="xxs"
                  />
                </span>
              </osds-link>
            </div>
          </osds-tile>
        </a>
      </aside>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
